### PR TITLE
feat: datetime to dateonly and timeonly mapping

### PR DESCRIPTION
**Pull request by justi7n**
Closes https://github.com/riok/mapperly/issues/243

### CODE REVIEW

File: src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            return null;
+
+        if (ctx.Target is not INamedTypeSymbol namedSymbol || !SymbolEqualityComparer.Default.Equals(namedSymbol.ConstructedFrom, ctx.Types.DateOnly))
+            return null;
+
+        var fromDateTimeMethod = ResolveFromDateTimeMethod(ctx);
+        if (fromDateTimeMethod is null)
+            return null;
+
+
```
Comment (latonz) [inline src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs:26]: Two NLs

File: src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToTimeOnlyMappingBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            return null;
+
+        if (ctx.Target is not INamedTypeSymbol namedSymbol || !SymbolEqualityComparer.Default.Equals(namedSymbol.ConstructedFrom, ctx.Types.TimeOnly))
+            return null;
+
+        var fromDateTimeMethod = ResolveFromDateTimeMethod(ctx);
+        if (fromDateTimeMethod is null)
+            return null;
+
+
```
Comment (latonz) [inline src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToTimeOnlyMappingBuilder.cs:26]: Two NL's

**Review by latonz — CHANGES_REQUESTED** [review]
Thank you for your contribution 😊
Could you include the new conversions in `11-conversions.md`?

File: src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    public static StaticMethodMapping? TryBuildMapping(MappingBuilderContext ctx)
+    {
+        if (!ctx.IsConversionEnabled(MappingConversionType.DateTimeToDateOnly) || ctx.Types.DateOnly == null)
+            return null;
+
+        if (ctx.Source.SpecialType != SpecialType.System_DateTime)
+            return null;
+
+        if (ctx.Target is not INamedTypeSymbol namedSymbol || !SymbolEqualityComparer.Default.Equals(namedSymbol.ConstructedFrom, ctx.Types.DateOnly))
```
Comment (latonz) [inline src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs:19]: ConstructedFrom isn't needed, is it?

File: src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs (hunk 1 above)
Comment (justi7n) [reply src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs]: Fixed

File: src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToTimeOnlyMappingBuilder.cs (hunk 2 above)
Comment (justi7n) [reply src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToTimeOnlyMappingBuilder.cs]: Fixed

File: src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs (hunk 3 above)
Comment (justi7n) [reply src/Riok.Mapperly/Descriptors/MappingBuilders/DateTimeToDateOnlyMappingBuilder.cs]: Right, fixed in both builders

**Review by latonz — APPROVED** [review]

**Pull request merged**